sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -208,18 +226,37 @@ def send_batch(self, event_data_batch, **kwargs):
                 :caption: Sends event data
 
         """
+        partition_id = kwargs.get("partition_id")
+        partition_key = kwargs.get("partition_key")
+        if isinstance(event_data_batch, EventDataBatch):
+            if partition_id or partition_key:
+                raise TypeError("partition_id and partition_key should be None when sending an EventDataBatch "
+                                "because type EventDataBatch itself may have partition_id or partition_key")
+            to_send_batch = event_data_batch
+        elif isinstance(event_data_batch, List):
```

Comment:
I wonder if it's better to simply try/except on TypeError when attempting iteration; since someone could pass a generator or class implementing a List as well.

Reply:
send_batch(List) only allows a finite list now. Generator/Iterator is deliberately not supported.
If a user has a generator or iterator, they should use EventDataBatch. The List is only used when they're sure the whole List is under size limit.

Reply:
Removed checking type List as per our offline discussion